src/span-data.ts

```diff
@@ -42,8 +42,15 @@ interface StackFrame {
   column_number?: number;
 }
 
-// Auto-incrementing integer
-let uid = 1;
+// Span IDs in the X-Cloud-Trace-Context header are 64-bit unsigned integers
+// This is the largest JavaScript number that is less than the max unsigned
+// 64-bit int, that is less than 2^64-1.
+const SPAN_ID_MAX = 18446744073709550000;
+
+function randomSpanId() {
+  // No rounding is needed because SPAN_ID_MAX is bigger than 1 / Number.EPSILON
+  return String(Math.random() * SPAN_ID_MAX);
```

Comment:
It might be in practice that `Math.random()` generates numbers larger than `Number.EPSILON` but I don't think this is necessarily guaranteed, as there can certainly be numbers smaller than `Number.EPSILON` (take Number.MIN_VALUE for example).

I did an experiment [here](https://gist.github.com/kjin/321ce5514a0a393f35c6aa71ef038789) and observed some interesting results:

* Stringifying a number is probably the bottleneck, not rounding
* `crypto.randomFillSync` is ~5 times slower for 10000000 requests

But given that even each `randomFillSync` call takes about ~2 microseconds we may be over-optimizing in the first place. My initial comment on whether we should use `crypto` was less about saying that we should use `Math.random`, and more wondering about whether the slowdown justified the increase in randomness. Given the uncertainty about how multiplying a decimal number by a large value gives a uniform distribution of integers (at least it's uncertain in my mind) it may actually be better to use `crypto.randomFillSync`. What do you think? (Sorry if it seems like I'm flip-flopping, that wasn't my intention)

Reply:
So `Math.random()` generates numbers between 0 and 1, so it almost always generates numbers larger than `Number.EPSILON`. Number.MIN_VALUE is negative. You're right though in that the distribution won't cover all possible numbers as by multiplying by Math.random though.

Thanks for benchmarking `randomFillSync` and good to know that it's not to slow. I'll switch it back to that.

I think ideally we would pull in a 64-bit integer library for Node and then generate all 8 random bytes, but I think 6 random bytes is OK (even given that the `number` double precision floats don't represent all of them exactly so it will be less than 6 bytes of randomness I think).

But all in all, much more random than an incrementing counter this is replacing.

Reply:
OK, update done. I guess to clarify, I'm not saying we should for sure pull in an 64-bit integer library and do the full 8-bytes of crypto randomness (or even pull in a separate random number library!), because I think the 6-bytes of crypto randomness with some rounding error is fine overall and better than more dependencies.

Reply:
Sounds good. For future reference -- `Number.MIN_VALUE = 5e-324`, the smallest positive number representable in JS.

Reply:
Sorry - my bad on that. I was confused by the negative exponent I think and was not looking as deeply into this as I could have. Thanks for clarifying!